crates/compiler/unify/src/unify.rs

```diff
@@ -2533,7 +2533,7 @@ enum Rec {
     None,
     Left(Variable),
     Right(Variable),
-    Both(Variable, Variable),
+    Both(Variable),
```

Comment:
this is extremely confusing. I think it should just be `allow(dead_code)` here? unless @ayazhafiz agrees that this can be removed? but then the variant name should probably change.